modules/dxp/apps/saml/saml-impl/src/main/java/com/liferay/saml/runtime/internal/servlet/filter/SamlSameSiteLaxCookiesFilter.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.saml.runtime.internal.servlet.filter;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.ResourceBundleLoader;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+
+import java.io.PrintWriter;
+
+import java.util.ResourceBundle;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Stian Sigvartsen
+ */
+@Component(
+	immediate = true,
+	property = {
+		"before-filter=Session Id Filter", "dispatcher=REQUEST",
+		"init-param.url-regex-ignore-pattern=^/html/.+\\.(css|gif|html|ico|jpg|js|png)(\\?.*)?$",
+		"servlet-context-name=",
+		"servlet-filter-name=SAML SameSite Lax Support Filter",
+		"url-pattern=/c/portal/saml/acs", "url-pattern=/c/portal/saml/slo",
+		"url-pattern=/c/portal/saml/sso"
+	},
+	service = Filter.class
+)
+public class SamlSameSiteLaxCookiesFilter extends BaseSamlPortalFilter {
```

Comment:
Don't we want to make it configurable? Some deployments don't need this filter because they have "same-site" setup.

Reply:
IIRC the previous pull was including a new System Settings entry too.

Reply:
Hi @lipusz. This pull is a very different approach for fixing the issue specifically for SAML, unlike the previous pull.

In "same-site" setups the filter becomes pass-through when the user has an active session on the portal the filter is executing on. If they don't, then it will introduce 1 more request in the flow, but will have no detrimental effect really.